lib/reverse_words.rb

```diff
@@ -1,6 +1,51 @@
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
+#b: beginning, e: end
+def reverse_str(my_words, b, e)
+  while b < e
+    temp_letter = my_words[b]
+    my_words[b] = my_words[e]
+    my_words[e] = temp_letter
+    b += 1
+    e -= 1
+  end
+  return my_words
+end
+
+def is_punctuation?(c)
+  punc_str = " ><!@#$%^&*()-_\\|:;?\"'"
+  punc_str.each_char do |char|
+    if c == char
+      return true
+    end
+  end
+  return false
+end
+
+p is_punctuation?(" ")
```

Comment:
It's a good idea to remove these sorts of statements you used for debugging prior to commit. If you want to keep them around, it's fine to create a new file and put them there.